spring-boot/admin/src/main/resources/templates/admin/fragments/wizard/wizard.html

```diff
@@ -1,348 +1,389 @@
+            $("#amountDiv").attr('style', 'display: block');
+        });
+        // This function will check if the user inserted an input in the service value
+        $("#serviceCost").change(function () {
+            $("#serviceCost").val() == '' ? $("#clickNextButtonMessage").hide() : $("#clickNextButtonMessage").show();
+        });
+    });
+
+    var smartWizard = null;
+
+    function init_SmartWizard() {
+
+        if (typeof ($.fn.smartWizard) === 'undefined') {
+            return;
+        }
+
+        smartWizard = $('#wizard').smartWizard(
             {
-                    $("#serviceNameSummary").html($("#serviceName").val());
-                    $("#serviceUrlSummary").html($("#serviceUrl").val());
-
-                    if ($("#monthlyChargeInput").val() == "true")
-                    {
-                        $("#costMethod").html("Monthly charge")
-                    }
-
-                    if ($("#monthlyChargeInput").val() == "true")
-                    {
-                        $("#costMethod").html("Pay per month, " + $("#serviceCost").val() + " € per month")
-                    }
-
-                    if ($("#percallChargeInput").val() == "true")
-                    {
-                        $("#costMethod").html("Pay per number of calls, " + $("#serviceCost").val() + " € per 1000 calls")
-                    }
-
-                    if ($("#serviceForFree").val() == "true")
-                    {
-                        $("#costMethod").html("Service For Free")
-                    }
+                autoAdjustHeight: true,
+                keyNavigation: false,
+                onLeaveStep: leaveAStepCallback,
+                onShowStep: onShowStepEvent,
+                onFinish: onFinishCallback
+            }
+        );
+
+        $('.buttonNext').addClass('btn btn-success');
+        $('.buttonPrevious').addClass('btn btn-primary');
+        $('.buttonFinish').addClass('btn btn-default');
+
+    }
+
+    function leaveAStepCallback(obj, context) {
+        if (context.toStep > context.fromStep)
+            return validateSteps(context.fromStep);
+        else
+            return true;
+    }
+
+    function validateSteps(stepnumber) {
+        //Step 1 and 5 don't need any validation.
+        switch (stepnumber) {
+            case 2:
+                return validateServiceName(stepnumber);
+                break;
+            case 3:
+                return validateURL(stepnumber);
+                break;
+            case 4:
+                return validateServiceCharge(stepnumber)
+                break;
+            default:
                 return true;
-
+        }
+    }
+
+    function validateServiceName(stepnumber) {
+        if ($("#serviceName").val().trim() == "") {
+            alert("Service name is mandatory");
+            return false;
+        }
+        return true;
+    }
+
+    function validateURL(stepnumber) {
+        if (stepnumber > 2) {
```

Comment:
This stepnumber check seems redundant, it is already checked in the switch statement. So this method also does not need the parameter `stepnumber` as far as I can see.

Reply:
done